### PR TITLE
Add pauli expansion to three qubit gate and update doc

### PR DESCRIPTION
**Pull request by dabacon**

### CODE REVIEW

File: cirq/ops/three_qubit_gates.py [hunk 1]
```diff
@@ -161,14 +161,21 @@ class ThreeQubitDiagonalGate(gate_features.ThreeQubitGate):
     def __init__(self,
                  diag_angles_radians: List[Union[float, sympy.Basic]]) -> None:
         """
+        A 3 qubit gate whose off-diagonal elements are zero and who on-diagonal
+        elements are all phases.
 
         Args:
             diag_angles_radians: The list of angles on the diagonal in radians.
-
+                If these values are $(x0, x1, \ldots , x^7)$ then the unitary
```
Comment (viathor) [inline cirq/ops/three_qubit_gates.py]: nit: Caret seems wrong here, you probably meant `x0, x1, \ldots, x7` or `x_0, x_1, \ldots, x_7`.

File: cirq/ops/three_qubit_gates.py [hunk 2]
```diff
@@ -161,14 +161,21 @@ class ThreeQubitDiagonalGate(gate_features.ThreeQubitGate):
...
+                has on diagonal values $(e^{i x0}, e^{i x1}, \ldots, e^{i x_7$.
```
Comment (viathor) [inline cirq/ops/three_qubit_gates.py]: nit: LaTex expression is not well-formed: curly brace and parenthesis are not closed. Also x0, x1 are not consistent with x_7 (either use underscore for all or none of the variables).

File: cirq/ops/three_qubit_gates.py [hunk 3]
```diff
@@ -161,14 +161,21 @@ class ThreeQubitDiagonalGate(gate_features.ThreeQubitGate):
...
         """
         self._diag_angles_radians = diag_angles_radians \
         # type: List[Union[float, sympy.Basic]]
 
+
+
+
+
```
Comment (viathor) [inline cirq/ops/three_qubit_gates.py]: nit: Too much vertical white-space.

File: cirq/ops/three_qubit_gates_test.py [hunk 4]
```diff
@@ -30,12 +30,14 @@ def test_eigen_gates_consistent_protocols(eigen_gate_type):
             eigen_gate_type, ignoring_global_phase=True)
 
 
-def test_consistent_protocols():
-    cirq.testing.assert_implements_consistent_protocols(cirq.CSWAP)
-
+@pytest.mark.parametrize(
+    'gate,ignoring_global_phase',
+    ((cirq.CSWAP, False),
+     (cirq.ThreeQubitDiagonalGate([2, 3, 5, 7, 11, 13, 17, 19]), True),
```
Comment (viathor) [inline cirq/ops/three_qubit_gates_test.py]: Why do we need to ignore overall phase here?

Comment (dabacon) [reply cirq/ops/three_qubit_gates_test.py]: The decompose doesn't respect global phase.

File: cirq/ops/three_qubit_gates.py (hunk 3 above)
Comment (dabacon) [reply cirq/ops/three_qubit_gates.py]: losing fight with formatter

fixed

File: cirq/ops/three_qubit_gates.py (hunk 2 above)
Comment (dabacon) [reply cirq/ops/three_qubit_gates.py]: fixed

File: cirq/ops/three_qubit_gates.py (hunk 1 above)
Comment (dabacon) [reply cirq/ops/three_qubit_gates.py]: fixed

**Pull request merged**